### PR TITLE
[autograd] static Jacobian pass tests.

### PR DESCRIPTION
**Pull request by tongxin**
### PR types
New features

### PR changes
APIs

### Describe
Adds declarative Jacobian API.

*Usage*

```
from paddle.autograd.functional import Jacobian

def f(x):
   return x*x

with fluid.program_guard(main, startup):
    xs = make_tensors(inps)
    JJ = paddle.autograd.functional.Jacobian(pd_f, xs, batch=False)
    full_jacobian = JJ[:]
place = fluid.CUDAPlace(0)
    exe = fluid.Executor(place)
    exe.run(startup)
pd_jacobians = exe.run(main, feed=feeds, fetch_list=[full_jacobian])[0]
```

### CODE REVIEW

File: python/paddle/autograd/functional.py [hunk 1]
```diff
@@ -904,3 +904,73 @@ def func(x, y):
...
+    The Jacobian matrix of muli-input multi-output function.
+    """
+    def __init__(self, func, xs, batch=False):
+        r"""Requiring batch always take the 0'th axis both inputs and outputs. """
+        def enable_grads(xs):
+            if isinstance(xs, list):
+                for x in xs:
+                    x.stop_gradient = False
+            else:
+                assert isinstance(xs, paddle.fluid.framework.Variable)
```
Comment (levi131) [inline python/paddle/autograd/functional.py]: 建议assert 后边加上一些报错信息

File: python/paddle/fluid/tests/unittests/autograd/test_jacobian_static.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        paddle.enable_static()
+        self.np_dtype = np.float32
+        self.A = np.array([[1., 2.]]).astype('float32')
+        self.B = np.array([[1., 2.], [2., 1.]]).astype('float32')
+        self.C = np.array([[2., 2.], [2., 1.]]).astype('float32')
+        self.D = np.array([[[2., 2.], [2., 1.]], [[1., 2.], [2., 1.]]]).astype('float32')
+        self.E = np.array([[[3., 4.], [2., 3.]], [[2., 1.], [1., 3.]]]).astype('float32')
+        self.eps = 1e-4
+        self.rtol = 1e-3
+        self.atol = 1e-3
```
Comment (levi131) [inline python/paddle/fluid/tests/unittests/autograd/test_jacobian_static.py]: 建议增加一组float64数据类型的测试

File: python/paddle/autograd/functional.py (hunk 1 above)
Comment (tongxin) [reply python/paddle/autograd/functional.py]: 好的。

File: python/paddle/fluid/tests/unittests/autograd/test_jacobian_static.py (hunk 2 above)
Comment (tongxin) [reply python/paddle/fluid/tests/unittests/autograd/test_jacobian_static.py]: 谢谢提醒~

File: python/paddle/autograd/functional.py [hunk 3]
```diff
@@ -904,3 +904,73 @@ def func(x, y):
         vhp = grad_fn(jac, xs, v)
         outputs, vhp = return_fn(outputs), return_fn(vhp)
     return outputs, vhp
+
+
+from paddle.static import gradients
+class Jacobian(object):
```
Comment (veyron95) [inline python/paddle/autograd/functional.py]: Maybe the "from xx import xx" statement should be declared in the begging of this file?

Comment (tongxin) [reply python/paddle/autograd/functional.py]: Done.

**Review by levi131 — APPROVED** [review]

**Comment by levi131** [comment]
LGTM

**Pull request merged**